### ISSUE TITLE
windows doesnt bootstrap after themes change

### ISSUE THREAD

**Issue by erg** [factor/factor#1625]
```
Loading resource:basis/help/apropos/apropos.factor
Loading resource:basis/tools/completion/completion.factor
Loading resource:basis/help/vocabs/vocabs.factor
Loading resource:basis/help/home/home.factor
Loading resource:basis/ui/gadgets/borders/borders.factor
Loading resource:basis/ui/gadgets/editors/editors.factor
Loading resource:basis/documents/elements/elements.factor
Loading resource:basis/ui/gadgets/line-support/line-support.factor
Loading resource:basis/ui/gadgets/scrollers/scrollers.factor
Loading resource:basis/models/product/product.factor
Loading resource:basis/models/range/range.factor
Loading resource:basis/ui/gadgets/frames/frames.factor
Loading resource:basis/ui/gadgets/grids/grids.factor
Loading resource:basis/ui/gadgets/sliders/sliders.factor
Loading resource:basis/ui/gadgets/buttons/buttons.factor
Loading resource:basis/ui/gadgets/labels/labels.factor
Loading resource:basis/ui/gadgets/theme/theme.factor
Loading resource:basis/ui/images/images.factor
Loading resource:basis/images/loader/loader.factor
Loading resource:basis/images/loader/gdiplus/gdiplus.factor
Loading resource:basis/mime/types/types.factor
Loading resource:basis/windows/gdiplus/gdiplus.factor
Loading resource:basis/windows/streams/streams.factor
Loading resource:basis/windows/com/wrapper/wrapper.factor
Loading vocab:bootstrap/bootstrap-error.factor
(U) Quotation: [ c-to-factor => ]
    Word: c-to-factor
(U) Quotation: [
        2dup [
            ~quotation~ dip
            dup terminated?>> ~quotation~ ~quotation~ if
        ] 2dip => rot [ 2drop ] [ wrong-values ] if
    ]
    Word: call-effect
(U) Quotation: [
        [ ~quotation~ dip dip ] dip => dup terminated?>>
        [ 2drop f ]
        [ dup in>> length swap out>> length check-datastack ] if
    ]
(U) Quotation: [ swap [ call ] dip => ]
    Word: dip
(U) Quotation: [ [ (get-catchstack) push ] dip call => (get-catchstack) pop* ]
(U) Quotation: [
        nano-count f parser-quiet? set-global
        default-image-name "output-image" set-global
        default-components "include" set-global
        "" "exclude" set-global
        strip-encodings (command-line) parse-command-line
        os windows? [ "windows" require ] when
        "staging" get "deploy-vocab" get or
        [ "stage2: deployment mode" print ]
        [ "debugger" require "listener" require "none" require ] if
        load-components => nano-count over -
        core-bootstrap-time set-global
        run-bootstrap-init nano-count swap -
        bootstrap-time set-global print-report "deploy-vocab" get
        [ "tools.deploy.shaker" run ] [
            "staging" get ~quotation~ ~quotation~ if
            f error set-global f original-error set-global
            f error-continuation set-global
            "output-image" get save-image-and-exit
        ] if
    ]
(U) Quotation: [ iterate-step => iterate-next (each-integer) ]
(U) Quotation: [ swap [ 2dip ] dip => ]
    Word: 3dip
(U) Quotation: [ swap [ dip ] dip => ]
    Word: 2dip
(U) Quotation: [ swap [ call ] dip => ]
    Word: dip
(O) Method: M\ vocab (require)
(O) Word: load-source
(U) Quotation: [
        "ui.tools" require =>
        { "ui.backend.cocoa" } "ui.backend.cocoa.tools" require-when
        "ui.tools.walker" require
    ]
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: USING:
(O) Word: use-vocab
(O) Word: load-vocab
(O) Method: M\ vocab (require)
(O) Word: load-source
(O) Word: parse-file
(O) Word: parse-stream
(O) Word: parse-fresh
(O) Word: (parse-lines)
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: execute-parsing
(O) Word: POSTPONE: :
(O) Word: (:)
(O) Word: parse-definition
(O) Word: (parse-until)
(O) Word: parse-until-step
(O) Word: no-word
(O) Word: throw-restarts
(O) Method: M\ object throw
(U) Quotation: [
        OBJ-CURRENT-THREAD special-object error-thread set-global
        current-continuation => error-continuation set-global
        [ original-error set-global ] [ rethrow ] bi
    ]
resource:basis/ui/tools/tools.factor

3: USING: memory system kernel vocabs.refresh ui.tools.operations
                                                                 ^
resource:basis/ui/tools/operations/operations.factor

3: USING: accessors combinators.short-circuit compiler.errors
9: ui.gestures ui.operations ui.tools.browser ui.tools.deploy
                                             ^
resource:basis/ui/tools/browser/browser.factor

3: USING: accessors arrays assocs classes colors.constants
8: ui.gadgets.borders ui.gadgets.editors ui.gadgets.editors.private
                                        ^
resource:basis/ui/gadgets/editors/editors.factor

 3: USING: accessors arrays assocs calendar colors.constants
10: ui.gadgets.line-support ui.gadgets.menus ui.gadgets.scrollers
                           ^
resource:basis/ui/gadgets/line-support/line-support.factor

3: USING: accessors arrays combinators fry kernel math
5: opengl sequences ui.gadgets ui.gadgets.scrollers
                                                   ^
resource:basis/ui/gadgets/scrollers/scrollers.factor

3: USING: accessors arrays combinators combinators.short-circuit
6: ui.gadgets.grids ui.gadgets.private ui.gadgets.sliders
                                                         ^
resource:basis/ui/gadgets/sliders/sliders.factor

3: USING: accessors assocs combinators fry kernel math math.order
4: math.vectors models models.range ui.gadgets ui.gadgets.buttons
                                                                 ^
resource:basis/ui/gadgets/buttons/buttons.factor

3: USING: accessors assocs colors combinators combinators.short-circuit
6: ui.gadgets.labels ui.gadgets.packs ui.theme ui.gadgets.theme ui.gadgets.world
s
                    ^
resource:basis/ui/gadgets/labels/labels.factor

3: USING: accessors arrays colors.constants combinators fonts fry
5: strings ui.baseline-alignment ui.gadgets ui.gadgets.theme
                                                            ^
resource:basis/ui/gadgets/theme/theme.factor

 9: : theme-font-colors ( gadget -- gadget )
10:     [ content-background >>background text-color >>foreground ] change-font
;
                                                                               ^

No word named ΓÇ£change-fontΓÇ¥ found in current vocabulary search path

The following restarts are available:

:1      Defer word in current vocabulary
:2      Load resource:basis/ui/gadgets/theme/theme.factor again
:3      Load resource:basis/ui/gadgets/labels/labels.factor again
4 :res  Load resource:basis/ui/gadgets/buttons/buttons.factor again
5 :res  Load resource:basis/ui/gadgets/sliders/sliders.factor again
6 :res  Load resource:basis/ui/gadgets/scrollers/scrollers.factor again
7 :res  Load resource:basis/ui/gadgets/line-support/line-support.factor again
8 :res  Load resource:basis/ui/gadgets/editors/editors.factor again
9 :res  Load resource:basis/ui/tools/browser/browser.factor again
10 :res  Load resource:basis/ui/tools/operations/operations.factor again
11 :res  Load resource:basis/ui/tools/tools.factor again
Factor 0.98 x86.64 (1769, heads/master-63410f62c8, Tue May  3 11:13:36 2016)
[Microsoft Visual C++ 190023506] on windows
IN: scratchpad
```

**Comment by mrjbq7** [comment]
Thats a weird error -- why would a ``font`` slot not be defined yet?

Also why is the encoding weird on that paste?

```
No word named ΓÇ£change-fontΓÇ¥ found in current vocabulary search path
```

**Comment by erg** [comment]
Maybe the font slot is defined but there's circularity?

On Mon, May 9, 2016 at 11:15 PM, John Benediktsson <notifications@github.com

**Comment by mrjbq7** [comment]
Yep, fixed in 0781fa5d007ef240810fff5e93fefb3dc5604d89.  Not sure yet why it's necessary, but this will get bootstrap fixed.
